src/registrar/views/portfolio_members_json.py

```diff
@@ -56,92 +55,25 @@ def get(self, request):
     def initial_permissions_search(self, portfolio):
         """Perform initial search for permissions before applying any filters."""
-        permissions = UserPortfolioPermission.objects.filter(portfolio=portfolio)
-        permissions = (
-            permissions.select_related("user")
-            .annotate(
-                first_name=F("user__first_name"),
-                last_name=F("user__last_name"),
-                email_display=F("user__email"),
-                last_active=Coalesce(
-                    Cast(F("user__last_login"), output_field=TextField()),  # Cast last_login to text
-                    Value("Invalid date"),
-                    output_field=TextField(),
-                ),
-                additional_permissions_display=F("additional_permissions"),
-                member_display=Case(
-                    # If email is present and not blank, use email
-                    When(Q(user__email__isnull=False) & ~Q(user__email=""), then=F("user__email")),
-                    # If first name or last name is present, use concatenation of first_name + " " + last_name
-                    When(
-                        Q(user__first_name__isnull=False) | Q(user__last_name__isnull=False),
-                        then=Concat(
-                            Coalesce(F("user__first_name"), Value("")),
-                            Value(" "),
-                            Coalesce(F("user__last_name"), Value("")),
-                        ),
-                    ),
-                    # If neither, use an empty string
-                    default=Value(""),
-                    output_field=CharField(),
-                ),
-                domain_info=ArrayAgg(
-                    # an array of domains, with id and name, colon separated
-                    Concat(
-                        F("user__permissions__domain_id"),
-                        Value(":"),
-                        F("user__permissions__domain__name"),
-                        # specify the output_field to ensure union has same column types
-                        output_field=CharField(),
-                    ),
-                    distinct=True,
-                    filter=Q(user__permissions__domain__isnull=False)  # filter out null values
-                    & Q(
-                        user__permissions__domain__domain_info__portfolio=portfolio
-                    ),  # only include domains in portfolio
-                ),
-                type=Value("member", output_field=CharField()),
-            )
-            .values(
-                "id",
-                "first_name",
-                "last_name",
-                "email_display",
-                "last_active",
-                "roles",
-                "additional_permissions_display",
-                "member_display",
-                "domain_info",
-                "type",
-            )
+        queryset = UserPortfolioPermissionModelAnnotation.get_annotated_queryset(portfolio)
```

Comment:
last comment for now, by look at the code on the left. Do you see how it constantly is mentioning permissions, which then immediately lets you understand what the goal is for each of the filters etc. Here the new code is so abstracted that the meaning of this line has been lost.